packages/ui/src/components/atoms/Button.tsx

```diff
@@ -35,123 +141,53 @@ const buttonBaseStyle = css<ButtonProps>`
-            linear-gradient(#fff 0 0);
-          -webkit-mask-composite: xor;
-          mask-composite: exclude;
-        }
-
-        &:hover::before {
-          background: ${props.theme.palette.silver} border-box;
-        }
-
-        position: relative;
-        border: none;
-        background: ${props.theme.palette.golden};
-        -webkit-background-clip: text;
-        -webkit-text-fill-color: transparent;
-        background-clip: text;
-        text-fill-color: transparent;
-
-        &:hover {
-          background: ${props.theme.palette.silver};
-          -webkit-background-clip: text;
-          -webkit-text-fill-color: transparent;
-        }
-      `;
-    if (props.variant === 'dashed')
-      return css`
-        border: 1px dashed #49433e;
-        background: transparent;
-      `;
-    if (props.variant === 'warning')
-      return css`
-        background: ${({ theme }) => theme.palette.warning};
-        border: 0.6px solid #ff3518;
-        border-radius: 6px;
-        color: #ffffff;
-        font-weight: 500;
-      `;
-    if (props.variant === 'text')
-      return css`
-        background: none;
-        outline: none;
-        border: none;
-        transition: none;
-        padding: 0;
-        &:hover {
-          filter: brightness(150%);
-          cursor: pointer;
-        }
-      `;
-    if (props.variant === 'none')
-      return css`
-        background: transparent;
-        border: none;
-        padding: 0;
-      `;
-    return '';
+    return buttonVariantCssMap[props.variant ?? 'primary'];
   }}
 `;
 
+const buttonSizeStyle = css<ButtonProps>`
+  ${props => buttonSizeMap[props.size ?? 'md']}
+`;
+
 const ButtonStyle = styled.button<ButtonProps>`
   cursor: pointer;
-  font-weight: 500;
-  font-size: 14px;
-  line-height: 21px;
-  border-radius: 6px;
-  transition: all 0.15s ease-in-out;
-  display: inline-block;
-  padding-top: ${({ theme }) => theme.spacing.one.spacing};
-  padding-bottom: ${({ theme }) => theme.spacing.one.spacing};
-  padding-left: ${({ theme }) => theme.spacing.three.spacing};
-  padding-right: ${({ theme }) => theme.spacing.three.spacing};
-  ${spacing}
-  ${display}
-  ${buttonBaseStyle}
-  ${display}
-  ${width}
-  ${flex}
+  display: inline-flex;
+
+  ${buttonSizeStyle}
+  ${buttonStyle}
+
+  &:active {
+    box-shadow: 0px 0px 4px 4px rgba(0, 0, 0, 0.4) inset;
+    transition: box-shadow ${buttonAnimationData.duration}
+      ${buttonAnimationData.curve};
+  }
+
   ${utils}
 `;
 
-export const Button: FC<ButtonProps> = ({ children, ...props }) => (
-  <ButtonStyle type="button" {...props}>
-    {children}
-  </ButtonStyle>
-);
+export const Button: FC<ButtonProps> = ({
+  leadingIcon,
+  isLoading,
+  children,
+  ...props
+}) => {
+  const Leading = isLoading ? (
+    <Throbber size={throbberSizeMap[props.size ?? 'md']} strokeWidth={2} />
+  ) : (
+    leadingIcon
+  );
+  return (
+    <ButtonStyle type="button" disabled={isLoading} {...props}>
```

Comment:
```suggestion
    <ButtonStyle type="button" disabled={props.disabled ?? isLoading} {...props}>
```

Reply:
Added `isLoading || disabled` since either of those should be able to disable the button